addons/sale/report/sale_report.py

```diff
@@ -150,13 +151,20 @@ def _query(self, with_clause='', fields=None, groupby='', from_clause=''):
         if not fields:
             fields = {}
         with_ = ("WITH %s" % with_clause) if with_clause else ""
-        return '%s (SELECT %s FROM %s WHERE l.display_type IS NULL GROUP BY %s)' % \
+        sale_query = '%s (SELECT %s FROM %s WHERE l.display_type IS NULL GROUP BY %s)' % \
                (with_, self._select_sale(fields), self._from_sale(from_clause), self._group_by_sale(groupby))
+        return (sale_query, fields)
```

Comment:
We cannot change the API in stable, it would break any existing customization relying on `_query` returned value.